### PR TITLE
feat: remove context from adapters and pass it by parameter

### PR DESCRIPTION
**Pull request by whilefoo**
Before adapters were initialized by passing probot's context to the constructor but this a problem when multiple events are received simultaneously...the latest event overwrote the previous adapters so if any previous events were still processing and using logger it would post under wrong repository/issue

### CODE REVIEW

**Comment by whilefoo** [comment]
@pavlovcik

File: src/adapters/supabase/helpers/tables/access.ts [hunk 1]
```diff
@@ -19,15 +18,15 @@ type _Access = {
...
+  constructor(supabase: SupabaseClient) {
+    super(supabase);
   }
 
   private async _getUserWithAccess(id: number): Promise<UserWithAccess> {
     const { data, error } = await this.supabase.from("access").select("*, users(*)").filter("id", "eq", id);
 
     if (error) {
-      this.runtime.logger.error(error.message, error);
+      this.runtime.logger.error(null, error.message, error);
```
Comment (pavlovcik) [inline src/adapters/supabase/helpers/tables/access.ts:29]: Don't think you're using this correctly.

File: src/adapters/supabase/helpers/tables/logs.ts [hunk 2]
```diff
@@ -57,21 +58,17 @@ export class Logs extends Super {
...
+
+    if (context && postComment) {
+      const colorizedCommentMessage = this._diffColorCommentMessage(type, logMessage);
+      const commentMetaData = metadata ? Logs._commentMetaData(metadata, level) : null;
+      this._postComment(
+        context,
+        metadata ? [colorizedCommentMessage, commentMetaData].join("\n") : colorizedCommentMessage
+      );
     }
+
```
Comment (pavlovcik) [inline src/adapters/supabase/helpers/tables/logs.ts:71]: Logging works fine why did you change this

**Comment by pavlovcik** [comment]
> Before adapters were initialized by passing probot's context to the constructor but this a problem when multiple events are received simultaneously...the latest event overwrote the previous adapters so if any previous events were still processing and using logger it would post under wrong repository/issue

I don't understand how this is possible, because I thought you fixed this with your other pull request by passing in the context all the way from the program entry to the deepest nested functions.

File: src/adapters/supabase/helpers/tables/logs.ts (hunk 2 above)
Comment (pavlovcik) [reply src/adapters/supabase/helpers/tables/logs.ts:71]: It looks like you're removing logic around the comment getting colorized?

**Review by pavlovcik — CHANGES_REQUESTED** [review]
Change the function signature to be more developer experience friendly. More notes above.

**Comment by whilefoo** [comment]
> So it sounds like it is logger specific

it also affects other tables that use the probot context.

> It is a logger function so intuitively the log string should definitely be the first parameter. In addition, I see that you have instances where you are passing `null` as the `Context` object which means that it should be the later (and apparently optional) parameters.

I also wanted to put it as last optional parameter but the reason I put it as the first parameter is that you can't forget about it. If we put it as last optional parameter I think it will be often forgotten.

Another option is to have a logger inside the Context so you don't need to pass context and have Runtime as global logger where context is not available.

File: src/adapters/supabase/helpers/tables/logs.ts (hunk 2 above)
Comment (whilefoo) [reply src/adapters/supabase/helpers/tables/logs.ts:71]: I just simplified it

**Comment by pavlovcik** [comment]
> Another option is to have a logger inside the Context

We can override the built in one for probot possibly. I vaguely recall they have a deprecated logger property on the probot context object. 

I think that approach might be the cleanest.

File: src/adapters/index.ts [hunk 3]
```diff
@@ -21,7 +23,7 @@ export function createAdapters() {
       debit: new Settlement(supabaseClient),
       settlement: new Settlement(supabaseClient),
       label: new Label(supabaseClient),
-      logs: new Logs(supabaseClient, env.LOG_ENVIRONMENT, env.LOG_RETRY_LIMIT, env.LOG_LEVEL),
+      logs: new Logs(supabaseClient, null, env.LOG_ENVIRONMENT, env.LOG_RETRY_LIMIT, env.LOG_LEVEL),
```
Comment (pavlovcik) [inline src/adapters/index.ts:26]: I feel like params that are expected to be `null` should be last.

Comment (pavlovcik) [inline src/adapters/index.ts:26]: I feel like params that are expected to be `null` should be last.

File: src/handlers/comment/handlers/issue/relevance-scoring.ts [hunk 4]
```diff
@@ -135,7 +134,7 @@ interface InEachRequestParams {
 function filterSamples(context: Context, batchResults: number[][], correctLength: number) {
   return batchResults.filter((result) => {
     if (result.length != correctLength) {
-      Runtime.getState().logger.error(context.event, "Correct length is not defined", {
+      context.logger.error("Correct length is not defined", {
```
Comment (pavlovcik) [inline src/handlers/comment/handlers/issue/relevance-scoring.ts:137]: I might need to `throw` this error

**Review by pavlovcik — APPROVED** [review]
I trust that the code is stable so I'm merging it.

File: src/adapters/index.ts (hunk 3 above)
Comment (pavlovcik) [reply src/adapters/index.ts:26]: 0724395cfeb6ba4bce624bf41d99ff3d0f72ea00

**Pull request merged**